py/picca/delta_extraction/data_catalogues/desi_data.py

```diff
@@ -223,35 +223,45 @@ def read_from_healpix(self, catalogue):
         -----
         DataError if the analysis type is PK 1D and resolution data is not present
         """
-        in_nside = 64
+        in_nside = 16
+        is_mock = True
 
         healpix = [
             healpy.ang2pix(in_nside, np.pi / 2 - row["DEC"], row["RA"], nest=True)
             for row in catalogue
         ]
         catalogue["HEALPIX"] = healpix
         catalogue.sort("HEALPIX")
+
+        if not "SURVEY" in catalogue.colnames:
+            catalogue["SURVEY"]=np.ma.masked
+
         grouped_catalogue = catalogue.group_by(["HEALPIX", "SURVEY"])
 
         forests_by_targetid = {}
-        is_mock = True
+
         for (index,
              (healpix, survey)), group in zip(enumerate(grouped_catalogue.groups.keys),
                                     grouped_catalogue.groups):
-
-            input_directory = f'{self.input_directory}/{survey}/dark'
-            filename = (
+            input_directory = f'{self.input_directory}/'
+            filename_truth = (
+                f"{input_directory}/{healpix//100}/{healpix}/truth-{in_nside}-"
+                f"{healpix}.fits")
+            if not os.path.isfile(filename_truth):
```

Comment:
hmm I just noticed I have to change in_nside to 64 here for the data. But I wanted to check with @iprafols he agrees with this in order to read the data with the same DesiData type, or maybe he prefers to write a separate method even though the way it is working now it is this function the one indicating whether is mock or data...

Reply:
Hi @alma, I added a fix to the desi_extraction branch, I'm not sure I merged but it not I'll do it right now

Reply:
ok, so weirdly enough this was not set to 64 even when the merge was done. I will push a commit for this